Types/InsertChangeSet.php

```diff
@@ -66,21 +68,88 @@ public static function columns(array $tokens) {
      * @SuppressWarnings("PHPMD.StaticAccess")
      */
     public static function values(array $tokens) {
-        $values = explode(',', self::removeBrackets(self::removeSpacesBetweenComma(end($tokens['VALUES'])['base_expr'])));
+        $values = self::removeBrackets(end($tokens['VALUES'])['base_expr']);
 
         return array_map(function($value) {
             return Value::create($value);
-        }, $values);
+        }, static::parseValueStringToArray($values));
     }
 
     /**
-     * removes spaces between commas
+     * parses the values string into a an array that has the correct amount of values, compared with the column array.
      *
-     * @param  string $string
-     * @return string
+     * @see isCharStringDelimiter for a list of valid string starting characters.
+     * @see SqlQuery::quote() for the place where the automatic escape happens
+     *
+     * @param string $values
+     *
+     * @return array
+     */
+    private static function parseValueStringToArray($values) {
```

Comment:
Since this doesn't really apply to our naming policy I guess its from the patch?

Reply:
It's from the patch. 95% of things here are from it.